explainer.md

```diff
@@ -40,13 +47,105 @@ security researchers can check the transparency log for all published
 versions of a given web app, and check that none of them contain any
 malicious code.
 
+```
+# jka commentary
+A few observations here:
+
+  * A second channel to obtain the 'expected' content -- or a
+    cryptographically strong assertion that we can rely on to confirm whether
+    we've received matching/mismatching content is required with any approach.
+
+  * That second channel becomes an element in the chain-of-trust; we want it
+    to be as difficult as possible for potential attackers to disrupt, and as
+    easy as possible for our colleages / peers / security researchers to
+    verify independently.
+
+  * To some extent I'd argue that source control repositories (particularly
+    `git`, `fossil` and `hg`) provide some of the transparency and
+    cryptographic properties desired here.  However, there is no single
+    location where repositories are aggregated - and that may be a benefit
+    under some threat models.
```

Comment:
Essentially, this proposal is meant as a way to verify that the source code of a deployed web app is the same as the source code in the public repository (perhaps after some reproducible build step). So having a public repository of the source code is definitely an important (but not sufficient) step to achieve these goals.

Reply:
Maybe worth considering: does it make sense to provide some kind of verification signature for both the source code (something to do with the commit the source was built from?) and also the compiled artifact?

I'm not sure that would improve the security properties in an information-theoretical sense, but something like that could make it easier for people to find and check the source code version to build from.

All has to be relatively straightforward to automate for deployment purposes too I suppose.

Reply:
I think this is something that a participating web app could easily provide in a separate channel, e.g. somewhere on GitHub/GitLab/etc. That would definitely be useful to aid in auditing, if a build step was used. However, I don't think we should require it (as a web app may not have a build step at all, and could simply deploy the source code exactly as it's found on git).